fawkes/algorithms/algo.py

```diff
@@ -23,6 +23,55 @@
 from fawkes.review.review import Review
 from fawkes.cli.fawkes_actions import FawkesActions
 
+from sentence_transformers import SentenceTransformer
+import numpy as np
+from sklearn.cluster import KMeans
+from gensim.summarization.summarizer import summarize
+
+
+'''
+        @param{list<string>}: sentences - List of review sentence
+        @param{Integer}: num_clusters - User defined number of clusters
+        @returns{list<list<string>}: clustered_sentences - List of clusters of sentences 
+        
+        Groups similar sentences into clusters
+'''
+#Reference : https://github.com/UKPLab/sentence-transformers
+def k_means_classification(sentences, num_clusters):
+
+    #Loading pre-trained model - 'distilbert-base-nli-stsb-mean-tokens'
+    embedder = SentenceTransformer('distilbert-base-nli-stsb-mean-tokens')
+    corpus_embeddings = embedder.encode(sentences)
+   
+    #Apply k-means to groups similar sentences into 'num_clusters' clusters
+    clustering_model = KMeans(n_clusters=num_clusters)
+    clustering_model.fit(corpus_embeddings)
+    #assign label to each sentence
+    cluster_assignment = clustering_model.labels_
+
+    #group similar labeled sentences 
+    clustered_sentences = {new_list: [] for new_list in range(num_clusters)} 
+    for i in range(len(cluster_assignment)):
+        temp_list= clustered_sentences[cluster_assignment[i]]
+        temp_list.append(sentences[i])
+        clustered_sentences[cluster_assignment[i]]=temp_list
+   
+    return clustered_sentences
+
+'''
+        @param{string}: text - text corpus to summarize
+        @param{integer}: word_count - max number of words in a summary
+        @returns{list<string>}: generated summary
+        
+        function to summarize a piece of text
+'''
+def summarize_text(text,word_count):
```

Comment:
If summarisation is an algorithm, then the module should not write to a file on its own.

What algorithm does to modify the derived insights object for a Review.

I think you guys have mixed both the concepts.

Reply:
Did not understand this, summarize_text in algo.py is not writing into the file. Could you re-explain

Reply:
Why does summarize_text has to be algo.py ? If its not a pluggable algorithm that I can run in fawkes ?

Reply:
Ohh okay, i interpreted it the wrong way then. Will make the changes, Thanks!